### PR TITLE
amp-bind: Restore canBind()

### LINKED ISSUE #8271: amp-bind: Avoid attribute calculation by amp-mustache
`amp-mustache` can calculate arbitrary attributes, so we can't rely on the validator to constrain the set of bindable attributes.

E.g. `<a href="{{#x}}"/>{{/x}}http://example.com/x.html" [target]="_top" />`, where `x` is `false`.

### PR DESCRIPTION
**Pull request by choumx**
Fixes #8271.

- Partial revert of #8033.
- Restores `BindValidator.canBind()` with latest integrations.

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-validator.js [hunk 1]
```diff
@@ -33,9 +34,23 @@ let PropertyRulesDef;
  * @private {Object<string, ?PropertyRulesDef>}
  */
 const GLOBAL_PROPERTY_RULES = map({
+  'text': null,
```
Comment (kmh287) [inline extensions/amp-bind/0.1/bind-validator.js]: Do we want to allow any tag to have the `[text]` attribute? I could see this being potentially problematic on a `<div> `for instance, where reassigning its textContent could be used to remove its contents.

**Comment by kmh287** [comment]
Also, depending on which PR you submit first, don't forget to add amp-iframe to the rules map :)

File: extensions/amp-bind/0.1/bind-validator.js [hunk 2]
```diff
@@ -79,8 +106,13 @@ export class BindValidator {
       rules = this.rulesForTagAndProperty_(tag, rules.alternativeName);
     }
 
-    // If there are no rules governing this binding, return true.
-    if (!rules) {
+    // If binding to (tag, property) is not allowed, return false.
+    if (rules === undefined) {
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-validator.js]: This should be calling `#canBind`.

File: extensions/amp-bind/0.1/bind-validator.js (hunk 1 above)
Comment (choumx) [inline extensions/amp-bind/0.1/bind-validator.js]: Yea, I originally had only whitelisted text elements for `[text]` but I think a more general solution is more consistent (`Node.textContent` is not limited to text elements) and makes the validator changes smaller.

File: extensions/amp-bind/0.1/bind-validator.js (hunk 2 above)
Comment (choumx) [inline extensions/amp-bind/0.1/bind-validator.js]: It did at one point, but it causes a duplicate call to `rulesForTagAndProperty_`. This is slightly more efficient.

**Pull request merged**